File: Semana_2/Cypress/JavaScript/npm/curso/e2e/cypress/pages/separated/menu.actions.js
const { MENU_LOCATORS } = require('./menu.locators');
const { capture } = require('../../support/evidence');

class MenuActions {
  openNewAccount() {
    cy.get(MENU_LOCATORS.openNewAccount).click();
    capture('parabank-separated-open-new-account');
  }
}

module.exports = { MenuActions };

File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/separated/MenuActions.java
package com.curso.playwright.pages.separated;

import com.curso.playwright.evidence.EvidenceSupport;
import com.microsoft.playwright.Page;

public class MenuActions {
    private final Page page;
    private final MenuLocators locators;

    public MenuActions(Page page) {
        this.page = page;
        this.locators = new MenuLocators(page);
    }

    public void openNewAccount() {
        locators.openNewAccountLink().click();
        EvidenceSupport.capture(page, "parabank-separated-open-new-account");
    }
}
